File: handinhand-frontend/src/api/skill.ts
import { api } from './client';

export interface Skill {
  id: number;
  title: string;
  description?: string;
  category?: string;
  userId: number;
  userName?: string;
}

export const fetchSkills = (userId?: number) =>
  api.get<Skill[]>('/api/skills', { params: { userId } }).then(r => r.data);

export const createSkill = (s: { title: string; description?: string; category?: string }) =>
  api.post<Skill>('/api/skills', s).then(r => r.data);

export const mySkills = () => api.get<Skill[]>('/api/skills/me').then(r => r.data);
export const updateSkill = (s: Skill) => api.put(`/api/skills/${s.id}`, s);
export const deleteSkill = (id: number) => api.delete(`/api/skills/${id}`);

/* 详情 */
export const getSkill = (id: number) =>
  api.get<Skill>(`/api/skills/${id}`).then(r => r.data);

File: handinhand-backend/Models/Skill.cs
namespace HandInHand.Models;
public class Skill {
    public int Id { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    // 简单打标签，前端可做筛选
    public string? Category { get; set; }
    // 外键
    public int UserId { get; set; }
    // 导航属性
    public User? User { get; set; }

    public ICollection<Comment> Comments { get; } = new List<Comment>();
}